doc/api.md

```diff
@@ -955,6 +966,117 @@ The IOTA processes the entity attributes looking for a `TimeInstant` attribute.
+
+Push commands are only valid for HTTP devices. For MQTT devices it is not needed to set the `endpoint` parameter. 
+
+**Poll commands**
+
+Poll commands are those that are stored in the IoT Agent waiting to be retrieved by the devices. This kind of 
+commands are typically used for devices that doesn't have a public IP or the IP cannot be reached because of 
+power or netkork constrictions. The device connects to the IoT Agent periodically to retrieve commands. In order 
+to configure the device as poll commands you just need to ignore the `endpoint` parameter in the device provision.
+
+Once the command request is issued to the IoT agent, the command is stored waiting to be retrieved by the device. In that moment, the status of the command is `"command_status":"PENDING"`. 
+
+For HTTP devices, in order to retrieve a pull command from IoTA-JSON Agent the device should make the following request:
+
+```
+GET /iot/json?k=<apikey>&i=<deviceId>&getCmd=1
+Accept: application/json
+```
+
+For IoT Agents it is exactly the same just changing in the request the resource, `/iot/json` by the corresponding resource employed by the agent (I.E, IoTA-UL uses `/iot/d` as default resource) and setting the correct apikey and deviceId.
+
+It can be also possible for a device to retrieve the commands from the IoT Agent when it sends and observation. It just be needed to include the `&getCmd=1` parameter in the observation request. In the following example a device sends an JSON observation and retrieves the commands from the IoTA-JSON Agent.
+
+```
+POST /iot/json?k=<apikey>&i=<deviceId>&getCmd=1
+Content-Type: application/json
+
+{"t":25,"h":42,"l":"1299"}
+```
+
+This is also possible for IoTA-UL Agent changing in the request the resource, setting the correct apikey, deviceId, payload and headers.
+
+Once the command is retrieved by the device the status is updated to `"command_status":"DELIVERED"`.
+
+
+#### MQTT devices
+
+For MQTT devices, it is not needed to declare and endpoint. The device is supposed to be subscribed to the 
+following MQTT topic:
+
+```
+/<apiKey>/<deviceId>/cmd
+```
+
+where it will receive the command information. Please note that the device should subscribe to the broker 
+using the disabled clean session mode (enabled using `--disable-clean-session` option CLI parameter in 
+`mosquitto_sub`). This option means that all of the subscriptions for the device will be maintained after 
+it disconnects, along with subsequent QoS 1 and QoS 2 commands that arrive. When the device reconnects, it
+will receive all of the queued commands.
+
+### Command confirmation
+
+Once the command is complely procesed by the device, it should return the result of the command to the IoT 
+Agent. This result will be progressed to the Context Broker where it will be stored in the `command_info` 
+attribute. The status of the command will be stored in the `command_status` attribute (`OK` if everything 
+goes right). 
```

Comment:
```suggestion
Once the command is completely processed by the device, it should return the result of the command to the IoT 
Agent. This result will be progressed to the Context Broker where it will be stored in the `<command>_info` 
attribute. The status of the command will be stored in the `<command>_status` attribute (`OK` if everything 
goes right). 
```